nix/build-haskell-package.nix

```diff
@@ -22,15 +22,37 @@ let
       name = "${name}";
       inherit src;
     };
+
 in
 
-name: root:
+name: root: cabal2NixFile:
 lib.pipe root
   [
     # Avoid rebuilding because of changes in parent directories
     (mkNewStorePath "source-${name}")
     (x: log.traceDebug "${name}.mkNewStorePath ${x.outPath}" x)
 
-    (root: self.callCabal2nix name root { })
-    (x: log.traceDebug "${name}.cabal2nixDeriver ${x.cabal2nixDeriver.outPath}" x)
+    (root:
+      let path = lib.concatStringsSep "/" [ root "${cabal2NixFile}" ];
+      in
+      # Check if cached cabal2nix generated nix expression is present,
+        # if present use it with callPackage
+        # to avoid IFD
+      if lib.pathIsRegularFile path
+      then
+        (log.traceDebug "${name}.callPackage[Cached] ${path}")
+          self.callPackage
+          path
+          { }
+      else
+        let
+          pkg =
+            (log.traceDebug "${name}.callCabal2nix ${root}")
+              self.callCabal2nix
+              name
+              root
+              { };
+        in
+        (log.traceDebug "${name}.cabal2nixDeriver ${pkg.cabal2nixDeriver.outPath}" pkg)
```

Comment:
I think should only be logged when `callCabal2nix` is being used.

Reply:
Yeah. This is exactly that happens. 
` (log.traceDebug "${name}.cabal2nixDeriver ${pkg.cabal2nixDeriver.outPath}" pkg)` gets logged only when we use `callCabal2nix`.

Logs from `nammayatri`
https://gist.github.com/rsrohitsingh682/243463dbffaeea5fc87f488a997736d1